src/echo.ts

```diff
@@ -1,24 +1,29 @@
 import { Channel, PresenceChannel } from './channel';
 import { PusherConnector, SocketIoConnector, NullConnector } from './connector';
+import type { Connector, Options } from './connector';
+
+interface EchoOptions extends Options {
+    withoutInterceptors?: boolean;
+}
 
 /**
  * This class is the primary API for interacting with broadcasting.
  */
-export default class Echo {
+export default class Echo<O extends Record<PropertyKey, unknown>> {
```

Comment:
user can pass in the typings from providers such as [pusher](https://github.com/pusher/pusher-js/blob/7024cc8e1b68e5357b8ecacc9a0650b1033d6a07/src/core/options.ts#L13)